storage/storage_image.go

```diff
@@ -1207,7 +1207,7 @@ func (s *storageImageDestination) Commit(ctx context.Context, unparsedToplevel t
 		if len(oldNames) > 0 {
 			names = append(names, oldNames...)
 		}
-		if err := s.imageRef.transport.store.SetNames(img.ID, names); err != nil {
+		if err := s.imageRef.transport.store.AddNames(img.ID, names); err != nil {
 			logrus.Debugf("error setting names %v on image %q: %v", names, img.ID, err)
 			return errors.Wrapf(err, "setting names %v on image %q", names, img.ID)
 		}
```

Comment:
All of these texts should only list the newly added names now; they shouldn’t list a set of names that might never actually exist in storage.